compiler/executable/match_/planner/plan.rs

```diff
@@ -1208,11 +1218,9 @@ impl PartialCostPlan {
 
     fn hash(&self) -> PartialCostHash {
         PartialCostHash {
-            all_vars: self.remaining_patterns.clone(),
-            ongoing_vars: self.ongoing_step.clone(),
-            approx_io: (self.cumulative_cost.io_ratio * self.ongoing_step_cost.io_ratio) as u64, // TODO: improve rounding/hashing (make relative)
-            approx_cost: self.cumulative_cost.chain(self.ongoing_step_cost).cost as u64, // TODO: improve rounding/hashing (make relative)
-            ongoing_join: self.ongoing_step_join_var.is_some(),
+            n_remaining_patterns: self.remaining_patterns.len() as u32,
+            planned_vertices: self.vertex_ordering.iter().filter_map(|v| v.as_pattern_id()).collect::<BTreeSet<_>>(),
+            ongoing_non_trivial_patterns: self.ongoing_step.iter().copied().collect::<BTreeSet<_>>(),
```

Comment:
This needs to be checked for correctness.

Reply:
Is `n_remaining_patterns` meaningful? Surely `planned_patterns` + `ongoing_non_trivial_patterns` already account for all the information, and the stash doesn't really do anything by definition.
I also wonder if we can just lump them all together into `planned_patterns`. `ongoing_step_join_var` may disambiguate enough on its own.

Reply:
In a beam search, it's irrelevant because we reset the hash-set. 
In a search that doesn't proceed in steps, we need it so we can actually add things to the stash without it being instantly ruled out by the hash.
I think an A* will be helpful to evaluate our heuristic so Idm the num_patterns being in there.